drivers/i2c/i2c_ite_enhance.c

```diff
@@ -568,6 +605,311 @@ static int i2c_enhance_transfer(const struct device *dev, struct i2c_msg *msgs,
 	if (data->err || (msgs->flags & I2C_MSG_STOP)) {
 		data->i2ccs = I2C_CH_NORMAL;
 	}
+
+	return data->err;
+}
+
+#ifdef CONFIG_I2C_IT8XXX2_CQ_MODE
+static void i2c_enhance_enable_cpu_idle(int port)
```

Comment:
There are different power saving states in the EC including Doze(Idle), Deep Doze, and the Sleep modes. The EC processor(CPU) cannot be in the Idle/Doze mode during the transactions with the CQ mode(DMA mode). Otherwise, the EC processor would be clock gated. The functions pm_policy_state_lock_get(), pm_policy_state_lock_put() can only stop the EC processor from entering the Deep Doze, but it’s still possible that EC processor enters the Doze(Idle) mode.

Reply:
Please add a comment in this driver also describing the limitation of `k_cpu_idle()` and command-queue mode.

Reply:
I was added it at line 605, it seems more appropriate by moving it above. Done.

Reply:
Having the power code refer to this specific driver seems a little weird; could it be turned around? soc.c could export a pair of functions like `chip_block_idle` and `chip_permit_idle` that work the same as PM locks (atomic increment/decrement of a counter and skip `wfi` if the count is nonzero).

To avoid extra cost then there could be a SoC kconfig option that `CQ_MODE` selects to enable that chip API.

Reply:
> Having the power code refer to this specific driver seems a little weird; could it be turned around? soc.c could export a pair of functions like `chip_block_idle` and `chip_permit_idle` 

Done.

>that work the same as PM locks (atomic increment/decrement of a counter and skip `wfi` if the count is nonzero).

atomic_set_bit/atomic_clear_bit can also ensure that it will not enter `wfi`.

> To avoid extra cost then there could be a SoC kconfig option that `CQ_MODE` selects to enable that chip API.

Done.